.github/workflows/main.yml

```diff
@@ -18,50 +21,121 @@ jobs:
     - name: Checkout
       uses: actions/checkout@v4
       with:
-        submodules: true
+        submodules: false  # We'll handle this manually for better control
+        token: ${{ secrets.GITHUB_TOKEN }}
+    - name: Initialize Submodules (Robust)
+      run: |
+        echo "=== Initial Git Status ==="
+        git status
+        echo "=== Checking .gitmodules ==="
+        cat .gitmodules || echo "No .gitmodules found"
+        echo "=== Checking git config for submodules ==="
+        git config --list | grep submodule || echo "No submodule config found"
+        echo "=== Checking .git/modules ==="
+        ls -la .git/modules/ || echo "No .git/modules found"
+        echo "=== Force clean submodule state and re-init ==="
+        git submodule deinit --all --force || true
+        rm -rf .git/modules/library || true
+        rm -rf .git/modules/app || true
+        echo "=== Re-sync and init submodules ==="
+        git submodule sync --recursive
+        git submodule update --init --recursive --force
+        echo "=== Submodule status after clean init ==="
+        git submodule status --recursive
+        echo "=== Manual clone fallback if needed ==="
+        if [ ! -d "library/src/main/cpp/love" ]; then
+          echo "Love submodule still missing, manual clone..."
+          mkdir -p library/src/main/cpp
+          git clone https://github.com/love2d/love.git library/src/main/cpp/love --depth 1
+        fi
+        if [ ! -d "library/src/main/cpp/megasource" ]; then
+          echo "Megasource submodule still missing, manual clone..."
+          mkdir -p library/src/main/cpp
+          git clone https://github.com/love2d/megasource.git library/src/main/cpp/megasource --depth 1
+        fi
+        echo "=== Final verification ==="
+        ls -la library/src/main/cpp/
+        ls -la library/src/main/cpp/love/ | head -5 || echo "love directory empty/missing"
+        ls -la library/src/main/cpp/megasource/ | head -5 || echo "megasource directory empty/missing"
```

Comment:
Any particular reason why we have to clone the submodules manually? This breaks people who just want to compile quickly (see my above review).